src/main/java/com/myaudiolibrary/web/controller/ArtistController.java

```diff
@@ -0,0 +1,208 @@
+package com.myaudiolibrary.web.controller;
+
+
+import com.myaudiolibrary.web.model.Artist;
+import com.myaudiolibrary.web.repository.AlbumRepository;
+import com.myaudiolibrary.web.repository.ArtistRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.EntityExistsException;
+import javax.persistence.EntityNotFoundException;
+import java.util.Optional;
+
+@RestController
+@RequestMapping(value = "/artists")
+public class ArtistController
+{
+
+
+    @Autowired
+    private ArtistRepository artistRepository;
+
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                            RECHERCHE ARTISTE PAR ID                                                //
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    @RequestMapping(method = RequestMethod.GET,  produces = "application/json"  , value = "/{id}")
+    public Artist artistInf(@PathVariable(value = "id") Long id)
+    {
+        Optional<Artist> optionalArtist = artistRepository.findById(id);
+        if (optionalArtist.isEmpty())
+        {
+            throw new EntityNotFoundException("aucun artist id : " + id + " n'a été trouvé.");
+        }
+
+        Artist artist = optionalArtist.get();
+        return artist;
```

Comment:
```suggestion
        return optionalArtist.get();
```

De ce qu'on a vu en TP, il me semble que tu peux aussi directement renvoyer l'Optional, le JSON généré sera le même